api/alert_rules.go

```diff
@@ -0,0 +1,201 @@
+
+// String returns the string representation of a Alert Rule integration type
+func (i alertRuleType) String() string {
+	return string(i)
+}
+
+func (sevs AlertRuleSeverities) toInt() []int {
+	var res []int
+	for _, i := range sevs {
+		res = append(res, int(i))
+	}
+	return res
+}
+
+const (
+	AlertRuleSeverityCritical alertRuleSeverity = iota + 1
+	AlertRuleSeverityHigh
+	AlertRuleSeverityMedium
+	AlertRuleSeverityLow
+	AlertRuleSeverityInfo
+)
+
+// NewAlertRule returns an instance of the AlertRuleData struct
+//
+// Basic usage: Initialize a new AlertRuleData struct, then
+//              use the new instance to do CRUD operations
+//
+//   client, err := api.NewClient("account")
+//   if err != nil {
+//     return err
+//   }
+//
+//   alertRule := api.NewAlertRule(
+//		"Foo",
+//     api.AlertRule{
+//		Description: "My Alert Rule"
+//		Severities: api.AlertRuleSeverities{api.AlertRuleSeverityHigh,
+//		Channels: []string{"TECHALLY_000000000000AAAAAAAAAAAAAAAAAAAA"},
+//		ResourceGroups: []string{"TECHALLY_111111111111AAAAAAAAAAAAAAAAAAAA"}
+//       },
+//     },
+//   )
+//
+//   client.V2.AlertRules.Create(alertRule)
+//
+func NewAlertRule(name string, rule AlertRule) AlertRuleData {
+	return AlertRuleData{
+		Channels: rule.Channels,
+		Type:     AlertRuleEventType.String(),
+		Filter: AlertRuleFilter{
+			AlertRuleComputedData: AlertRuleComputedData{},
+			Name:                  name,
+			Enabled:               1,
+			Description:           rule.Description,
+			Severity:              rule.Severities.toInt(),
+			ResourceGroups:        rule.ResourceGroups,
+			EventCategories:       rule.EventCategories,
+		},
+	}
+}
+
+// List returns a list of Alert Rules
+func (svc *AlertRulesService) List() (response AlertRulesResponse, err error) {
+	err = svc.client.RequestDecoder("GET", apiV2AlertRules, nil, &response)
+	return
+}
+
+// Create creates a single Alert Rule
+func (svc *AlertRulesService) Create(rule AlertRuleData) (
+	response AlertRuleResponse,
+	err error,
+) {
+	err = svc.create(rule, &response)
+	return
+}
+
+// Delete deletes a Alert Rule that matches the provided guid
+func (svc *AlertRulesService) Delete(guid string) error {
+	if guid == "" {
+		return errors.New("specify an intgGuid")
+	}
+
+	return svc.client.RequestDecoder(
+		"DELETE",
+		fmt.Sprintf(apiV2AlertRuleFromGUID, guid),
+		nil,
+		nil,
+	)
+}
+
+// Update updates a single Alert Rule of the provided guid.
+func (svc *AlertRulesService) Update(data AlertRuleData) (
+	response AlertRuleResponse,
+	err error,
+) {
+	err = svc.update(data.Guid, data, &response)
+	return
+}
+
+// Get returns a raw response of the Alert Rule with the matching guid.
+func (svc *AlertRulesService) Get(guid string, response interface{}) error {
+	return svc.get(guid, &response)
+}
+
+type AlertRule struct {
+	Channels        []string
+	Description     string
+	Severities      AlertRuleSeverities
+	ResourceGroups  []string
+	EventCategories []string
+}
```

Comment:
I like this struct 💯 - I just feel it might be confusing between `AlertRuleData` and `AlertRule`. But I like it since the user can customize it directly and pass it to `NewAlertRule()`

Reply:
I've left the name as is for now.